exporter/elasticsearchexporter/integrationtests/exporter_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package integrationtests
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestExporter(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		restartCollector bool
+		mockESFailure    bool
+	}{
+		{name: "basic"},
+		{name: "es_intermittent_failure", mockESFailure: true},
+		/* Below tests should be enabled after #30792 is fixed
+		{name: "collector_restarts", restartCollector: true},
+		{name: "collector_restart_with_es_intermittent_failure", mockESFailure: true, restartCollector: true},
+		*/
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			runner(t, tc.restartCollector, tc.mockESFailure)
+		})
+	}
+}
+
+func runner(t *testing.T, restartCollector, mockESFailure bool) {
+	t.Helper()
+
+	cfg, err := loadConfig()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mockES, err := newMockESClient(t, cfg.Debug)
+	require.NoError(t, err)
+
+	collector, err := newTestCollector(t, cfg, mockES.ServerURL)
+	require.NoError(t, err)
+	t.Cleanup(collector.Shutdown)
+
+	var g errgroup.Group
+	cancelRun := runTestCollectorWithWait(ctx, t, collector, &g)
+
+	if mockESFailure {
+		mockES.SetReturnStatusCode(http.StatusServiceUnavailable)
+		if !restartCollector {
+			// If restartCollector is not set then recover ES service
+			// after a few failed bulk requests otherwise ensure that
+			// both collector and ES are unavailable for the same
+			// duration considering worst case scenario.
+			time.AfterFunc(10*time.Second, func() {
+				mockES.SetReturnStatusCode(http.StatusOK)
+			})
+		}
+	}
+
+	var totalLogCount int
+	sendLogs(t, cfg.GRPCEndpoint, "batch_1", 2_000 /* log count */, 10)
+	totalLogCount += 2_000
+
+	if restartCollector {
+		// Restart the collector after all data is sent to the collector.
+		// Note that the log count should be high to ensure that the collector
+		// stops before all data is shipped to Elasticsearch.
+		require.NoError(t, collector.Recreate())
+		cancelRun()
+		// Ensure that the previous collector instance is shut down and then
+		// run the recreated collector.
+		require.NoError(t, g.Wait())
+		runTestCollectorWithWait(ctx, t, collector, &g)
+		// Recover ES if it was unavailable.
+		mockES.SetReturnStatusCode(http.StatusOK)
+	}
+
+	sendLogs(t, cfg.GRPCEndpoint, "batch_2", 2_000 /* log count */, 10)
+	totalLogCount += 2_000
+
+	assert.Eventually(
+		t, func() bool {
+			resp, err := mockES.Count(mockES.Count.WithIndex(cfg.ESLogsIndex))
+			require.NoError(t, err)
```

Comment:
I think we should close the body eventually?
```suggestion
			require.NoError(t, err)
			defer resp.Body.Close()
```